flagcx/kernels/device_onesided_demo.cu

```diff
@@ -0,0 +1,59 @@
+#include "comm.h"
+#include "flagcx.h"
+#include "flagcx_kernel.h"
+#include "global_comm.h"
+
+FLAGCX_GLOBAL_DECORATOR void flagcxOnesidedSendKernel(const void *srcbuff,
+                                                      size_t srcOffset,
+                                                      size_t dstOffset,
+                                                      size_t signalOffset,
+                                                      size_t count,
+                                                      flagcxDataType_t datatype,
+                                                      int peer, void *fifoBuffer) {
+  int tid = threadIdx.x;
+  if (tid == 0) {
+    flagcxDevicePut(srcbuff, srcOffset, dstOffset, count, datatype, peer,
+                    fifoBuffer);
+    flagcxDeviceSignal(signalOffset, peer, fifoBuffer);
+    flagcxDeviceTerm(fifoBuffer);
+    flagcxDeviceWait(fifoBuffer);
+  }
+}
+
+FLAGCX_GLOBAL_DECORATOR void flagcxOnesidedRecvKernel(volatile uint64_t *waitAddr,
+                                                      uint64_t expectedValue,
+                                                      void *fifoBuffer) {
+  int tid = threadIdx.x;
+  if (tid == 0) {
+    int iter = 0;
+    while (__atomic_load_n(waitAddr, __ATOMIC_ACQUIRE) != expectedValue) {
+      if (iter < 10) {
+        __nanosleep(1 << iter);
+      } else {
+        __nanosleep(1024);
+      }
+      iter++;
+    }
+    flagcxDeviceTerm(fifoBuffer);
+    flagcxDeviceWait(fifoBuffer);
+  }
+}
+
```

Comment:
The following code are not supposed to live in this same file.
Please move them to somewhere else.